control-plane/pkg/autoscaler/keda/keda.go

```diff
@@ -198,11 +198,19 @@ func GenerateTriggerAuthentication(cg *kafkainternals.ConsumerGroup, secretData
 			ca := kedav1alpha1.AuthSecretTargetRef{Parameter: "ca", Name: secret.Name, Key: security.CaCertificateKey}
 			secretTargetRefs = append(secretTargetRefs, ca)
 
-			cert := kedav1alpha1.AuthSecretTargetRef{Parameter: "cert", Name: secret.Name, Key: security.UserCertificate}
-			secretTargetRefs = append(secretTargetRefs, cert)
+			if secret.Data[security.TlsCertificate] != nil && secret.Data[security.TlsKey] != nil {
+				cert := kedav1alpha1.AuthSecretTargetRef{Parameter: "cert", Name: secret.Name, Key: security.TlsCertificate}
+				secretTargetRefs = append(secretTargetRefs, cert)
 
-			key := kedav1alpha1.AuthSecretTargetRef{Parameter: "key", Name: secret.Name, Key: security.UserKey}
-			secretTargetRefs = append(secretTargetRefs, key)
+				key := kedav1alpha1.AuthSecretTargetRef{Parameter: "key", Name: secret.Name, Key: security.TlsKey}
+				secretTargetRefs = append(secretTargetRefs, key)
+			} else {
+				cert := kedav1alpha1.AuthSecretTargetRef{Parameter: "cert", Name: secret.Name, Key: security.UserCertificate}
+				secretTargetRefs = append(secretTargetRefs, cert)
+
+				key := kedav1alpha1.AuthSecretTargetRef{Parameter: "key", Name: secret.Name, Key: security.UserKey}
+				secretTargetRefs = append(secretTargetRefs, key)
+			}
```

Comment:
The implementation in `control-plane/pkg/security/secret.go` allows setting mixed keys while this one doesn't

for example:

```
tls.key: <...>
user.crt: <...>
```
would be valid for the function `sslConfig` in `control-plane/pkg/security/secret.go` and it would use:
- the value of `tls.key` for the private key 
- `user.crt` for the public key

On the other end in this logic we would reference:
- the value of `user.key` for the private key 
- the value of `user.crt` for the public key
which is not consistent with the other file, so we need to check and set values individually and fallback to `user*` keys when not found:

Something like this (and the same for the other key):
```golang
if secret.Data[security.TlsKey] != nil {
    key := kedav1alpha1.AuthSecretTargetRef{Parameter: "key", Name: secret.Name, Key: security.TlsKey}
    secretTargetRefs = append(secretTargetRefs, key)
} else {
    key := kedav1alpha1.AuthSecretTargetRef{Parameter: "key", Name: secret.Name, Key: security.UserKey}
    secretTargetRefs = append(secretTargetRefs, key)
}